src/olympia/files/tasks.py

```diff
@@ -57,13 +58,18 @@ def repack_fileupload(results, upload_pk):
     # developer, so it could be something else.
     if upload.path.endswith('.xpi'):
         try:
-            tempdir = extract_zip(upload.path)
-        except Exception:
+            tempdir = tempfile.mkdtemp()  # *not* on TMP_PATH, we want local fs
+            zip_file = SafeZip(source=upload.path)
+            zip_file.extract_to_dest(tempdir)
```

Comment:
nit: this seems like a place where a decorator would be useful so the folder is always tidied up at the end of the block.

Reply:
mmm maybe I should keep using `extract_zip` - which does encapsulate `extract_to_dest`, though without being a decorator, and instead have it accept an argument for the `tempdir`...

Reply:
Done in 03bc9af8919da3af3b647420c9c54401fbd6fc15